src/arrays.js

```diff
@@ -1,61 +1,167 @@
+  // newArray.push(element);
+  // return newArray;
+
+  const newArray = array.map(item => item);
+  newArray.push(element);
+  return newArray;
 };
 
 const removeNthElement = (index, array) => {
-  // your code here
+  // for (let i = 0; i < array.length; i++) {
+  //   if (i === index) array.splice(i, 1);
+  // }
+
+  array.splice(index, 1);
 };
 
 const numbersToStrings = numbers => {
-  // your code here
+  // const newArray = [];
+  // numbers.forEach(num => {
+  //   newArray.push(String(num));
+  // });
+  // return newArray;
+
+  return numbers.map(num => String(num));
 };
 
 const uppercaseWordsInArray = strings => {
-  // your code here
+  // const newArray = [];
+  // strings.forEach(string => {
+  //   newArray.push(string.toUpperCase());
+  // });
+  // return newArray;
+
+  return strings.map(string => string.toUpperCase());
 };
 
 const reverseWordsInArray = strings => {
-  // your code here
+  const newArray = [];
+  for (let i = 0; i < strings.length; i++) {
+    let newString = '';
+    for (let j = strings[i].length - 1; j >= 0; j--) {
+      newString += strings[i][j];
+    }
+    newArray.push(newString);
+  }
+  return newArray;
 };
 
 const onlyEven = numbers => {
-  // your code here
+  // const newArray = [];
+  // numbers.forEach(num => {
+  //   if (num % 2 === 0) newArray.push(num);
+  // });
+  // return newArray;
+
+  return numbers.filter(num => num % 2 === 0);
 };
 
 const removeNthElement2 = (index, array) => {
-  // your code here
+  // const newArray = [];
+  // for (let i = 0; i < array.length; i++) {
+  //   if (i !== index) newArray.push(array[i]);
+  // }
+  // return newArray;
+
+  return array.filter((item, i) => i !== index);
 };
 
 const elementsStartingWithAVowel = strings => {
-  // your code here
+  const vowels = ['a', 'e', 'i', 'o', 'u'];
+  return strings.filter(item => vowels.includes(item[0].toLowerCase()));
+  // try with regex ?
 };
 
 const removeSpaces = string => {
-  // your code here
+  //return string.replace(/ /gi, '');
+  return string.split(' ').join('');
 };
 
 const sumNumbers = numbers => {
-  // your code here
+  // reduce((accumulator, currentValue)=>{});
+  return numbers.reduce((total, num) => {
+    return total + num;
+  }, 0);
 };
 
 const sortByLastLetter = strings => {
-  // your code here
+  const newArray = [];
+  while (strings.length > 0) {
+    const index = findSmallestIndex(strings);
+    newArray.push(strings[index]);
+    strings.splice(index, 1);
+  }
+  return newArray;
+};
+// given an array of strings, return the index of the item with the smallest last letter (charcode)
+const findSmallestIndex = arr => {
+  let smallestIndex = 0;
+  let smallestCharCode = arr[0].charCodeAt(arr[0].length - 1);
+  for (let i = 1; i < arr.length; i++) {
+    if (arr[i].charCodeAt(arr[i].length - 1) < smallestCharCode) {
+      smallestIndex = i;
+      smallestCharCode = arr[i].charCodeAt(arr[i].length - 1);
+    }
+  }
+  return smallestIndex;
```

Comment:
You can simplify this solution by using the `.sort` inbuilt array method. The final result looks something like:
```
  return strings.sort(
    (a, b) => a.charCodeAt(a.length - 1) - b.charCodeAt(b.length - 1)
  );
```